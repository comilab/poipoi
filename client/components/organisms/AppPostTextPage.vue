<script lang="ts">
import { defineComponent } from 'nuxt-composition-api'

import Post from '~/models/Post'
import usePostRenderer from '~/composables/use-post-renderer'
import AppSection from '~/components/atoms/AppSection.vue'

export default defineComponent({
  components: {
    AppSection
  },
  props: {
    post: {
      type: Post,
      required: true
    },
    page: {
      type: String,
      required: true
    }
  },
  render (createElement) {
    const { split } = usePostRenderer(this, createElement)
    const children = split(this.$props.page, [
      'break',
      'chapter',
      'image',
      'jump',
      'ruby',
      'jumpuri'
    ])
    return createElement(AppSection, {
      class: 'text-page'
    }, children)
  }
})
</script>

<style lang="postcss" scoped>
.text-page {
  @apply leading-loose;

  & h3 {
    @apply text-lg font-bold;
  }

  & a {
    @apply text-blue-500 transition-colors duration-300 cursor-pointer;

    &:hover {
      @apply text-blue-700 underline;
    }
  }
}
</style>
