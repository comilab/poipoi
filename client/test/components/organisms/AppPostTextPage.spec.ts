import { render } from '@testing-library/vue'

import AppPostTextPage from '~/components/organisms/AppPostTextPage.vue'
import imageFactory from '~/test/factories/image'
import postFactory from '~/test/factories/post'

describe('components/organisms/AppPostTextPage', () => {
  let options: any

  beforeEach(() => {
    options = {
      props: {
        post: postFactory.build(),
        page: ''
      },
      stubs: {
        NuxtLink: true
      }
    }
  })

  test('renders correctly', () => {
    options.props.post = postFactory.build({
      text: `
        [chapter:1ページ目の章タイトル]
        本文
        本文
        本文
        [jump:2]
        [[jumpuri:https://example.com>リンク]]
        [[rb:ルビ>るび]]
        [image:1]
        [newpage]
        [chapter:2ページ目の章タイトル]
      `,
      images: imageFactory.buildList(1, {
        publicPaths: {
          original: 'original',
          large: 'large',
          medium: 'medium',
          small: 'small'
        }
      })
    })
    options.props.page = `
      [chapter:1ページ目の章タイトル]
      本文
      本文
      本文
      [jump:2]
      [[jumpuri:https://example.com>リンク]]
      [[rb:ルビ>るび]]
      [image:1]
    `
    const { container } = render(AppPostTextPage, options)

    expect(container).toMatchSnapshot()
  })
})
