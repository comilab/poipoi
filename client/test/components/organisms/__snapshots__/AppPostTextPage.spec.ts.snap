// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/AppPostTextPage renders correctly 1`] = `
<div>
  <section
    class="rounded bg-white my-8 p-4 text-page"
    data-testid="section"
  >
    <!---->
    <br />
          
    <h3>
      1ページ目の章タイトル
    </h3>
    <br />
          本文
    <br />
          本文
    <br />
          本文
    <br />
          
    <a>
      2ページ目へ
    </a>
    <br />
          
    <a
      href="リンク"
      target="_blank"
    >
      https://example.com
    </a>
    <br />
          
    <ruby>
      ルビ
      <rt>
        るび
      </rt>
    </ruby>
    <br />
          
    <div
      class="inline-block"
    >
      <div
        class="zoomable-area relative"
        data-testid="zoomable-area"
      >
        <img
          class="select-none pointer-events-none"
          data-testid="zoomable-area-img"
          src="undefinedlarge"
        />
        <div
          class="caution absolute right-0 bottom-0 text-white text-xs p-2"
        >
          無断転載禁止
        </div>
      </div>
      <transition-stub
        enterclass="opacity-0"
        leavetoclass="opacity-0"
      >
        <!---->
      </transition-stub>
    </div>
    <br />
        
  </section>
</div>
`;
